hazelcast/src/main/java/com/hazelcast/internal/nio/tcp/TcpIpNetworkingService.java

```diff
@@ -256,20 +255,19 @@ public void scheduleDeferred(Runnable task, long delay, TimeUnit unit) {
     }
 
     private void startAcceptor() {
-        if (acceptor != null) {
-            logger.warning("TcpIpAcceptor is already running! Shutting down old acceptor...");
+        if (acceptorRef.get() != null) {
+            logger.warning("TcpIpAcceptor is already running! Shutting down old acceptorRef...");
             shutdownAcceptor();
         }
 
-        acceptor = new TcpIpAcceptor(registry, this, ioService).start();
-        metricsRegistry.collectMetrics(acceptor);
+        acceptorRef.set(new TcpIpAcceptor(registry, this, ioService).start());
     }
 
     private void shutdownAcceptor() {
+        TcpIpAcceptor acceptor = acceptorRef.get();
```

Comment:
The code is racy if multiple threads try to call `shutdownAcceptor`

Reply:
It's called from a synchronized method only, so currently it's not possible to call it concurrently.